### ISSUE TITLE
block-buffer: `BlockBuffer::get_pos()` with a block size of 0 can reach `unreachable_unchecked()`

### ISSUE THREAD

**Issue by LegionMammal978** [RustCrypto/utils#843]
This is effectively another instance of RustCrypto/traits#1275.
```rust
// RUSTFLAGS=-Cdebug-assertions=off cargo +nightly miri run
// block-buffer = "=0.10.3"
use block_buffer::{generic_array::typenum::U0, BlockBuffer, Eager};

fn main() {
    let buffer: BlockBuffer<U0, Eager> = Default::default();
    buffer.get_pos();
}
```
```rust
error: Undefined Behavior: entering unreachable code
   --> /home/lm978/.cargo/registry/src/github.com-1ecc6299db9ec823/block-buffer-0.10.3/src/lib.rs:195:17
    |
195 |                 core::hint::unreachable_unchecked();
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ entering unreachable code
    |
    = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
    = help: see https://doc.rust-lang.org/nightly/reference/behavior-considered-undefined.html for further information
    = note: BACKTRACE:
    = note: inside `block_buffer::BlockBuffer::<block_buffer::generic_array::typenum::UTerm, block_buffer::Eager>::get_pos` at /home/lm978/.cargo/registry/src/github.com-1ecc6299db9ec823/block-buffer-0.10.3/src/lib.rs:195:17: 195:52
note: inside `main`
   --> src/main.rs:7:5
    |
7   |     buffer.get_pos();
    |     ^^^^^^^^^^^^^^^^

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to previous error
```

**Comment by newpavlov** [comment]
Good catch! In v0.11 we introduce `BlockSizes` trait which gets [implemented](https://github.com/RustCrypto/traits/blob/master/crypto-common/src/lib.rs#L61-L78) for types from `U1` to `U255`. For v0.10 we probably should simply panic on attempted initialization with block size equal to zero.

**Comment by newpavlov** [comment]
Fixed in `block-buffer v0.10.4`.
